ee/hogai/graph/usage/queries.py

```diff
@@ -0,0 +1,280 @@
+                                msg -> JSONExtractArrayRaw(msg, 'tool_calls'),
+                                -- Only get messages from current turn (after last human message)
+                                arraySlice(
+                                    JSONExtractArrayRaw(
+                                        JSONExtractRaw(properties, '$ai_output_state'),
+                                        'messages'
+                                    ),
+                                    -- Start from the position after the last human message
+                                    arrayLastIndex(
+                                        x -> JSONExtractString(x, 'type') = 'human',
+                                        JSONExtractArrayRaw(
+                                            JSONExtractRaw(properties, '$ai_output_state'),
+                                            'messages'
+                                        )
+                                    ) + 1
+                                )
+                            )
+                        ) AS tool_calls
+                    FROM events
+                    PREWHERE
+                        team_id = %(team_to_query)s
+                        {region_filter}
+                        AND timestamp >= %(begin)s
+                        AND timestamp < %(end)s
+                        AND event = '$ai_trace'
+                )
+            ),
+            costs AS (
+                SELECT
+                    customer_team_id,
+                    trace_id,
+                    session_id,
+                    cost_usd
+                FROM (
+                    SELECT
+                        JSONExtractInt(properties, 'team_id') AS customer_team_id,
+                        JSONExtractString(properties, '$ai_trace_id') AS trace_id,
+                        JSONExtractString(properties, '$ai_session_id') AS session_id,
+                        toDecimal32OrNull(JSONExtractString(properties, '$ai_total_cost_usd'), 5) AS cost_usd,
+                        JSONExtractBool(properties, '$ai_billable') AS ai_billable
+                    FROM events
+                    PREWHERE
+                        team_id = %(team_to_query)s
+                        {region_filter}
+                        AND timestamp >= %(begin)s
+                        AND timestamp < %(end)s
+                        AND event = '$ai_generation'
+                )
+                WHERE
+                    ai_billable = 1
+                    AND cost_usd > 0
+                    AND cost_usd IS NOT NULL
+                    AND customer_team_id = %(team_id)s
+                    {session_filter}
+            )
+            SELECT toInt64(roundBankers(sum(c.cost_usd * 100 * %(markup_multiplier)s))) AS ai_credits
+            FROM costs c
+            LEFT JOIN trace_analysis t ON c.trace_id = t.trace_id AND c.session_id = t.session_id
+            WHERE t.is_billable = 1 OR t.trace_id IS NULL
```

Comment:
Looks like `{session_filter}` is the only difference from the usage report query (correct me if wrong!) If so, can these be the same query, just templatized? (It's totally fine to do `.format()` as there are no unsanitized client-provided strings here)